### PR TITLE
Make lowering refer to Core.Type (fixes #18756).

### LINKED ISSUE #18756: Cannot define type with name 'Type'
In wrapping LLVM (which provides provides classes named `Function`, `Module` and `Type`) I've been trying to create a non-exported type named `Type` in my package's namespace, for users to refer to by `LLVM.Type`. This fails (partially?):

```julia
julia> module Bar
       type Type
       end
       end
ERROR: too many parameters for type Type
 in eval_user_input(::Any, ::Base.REPL.REPLBackend) at ./REPL.jl:64
 in macro expansion at ./REPL.jl:95 [inlined]
 in (::Base.REPL.##3#4{Base.REPL.REPLBackend})() at ./event.jl:68

julia> Bar.Type
Bar.Type

julia> @which Type
Core
```

But does work for other types exported by Core, like `Module` or `Function`:

```julia
julia> module Foo
       type Module
       end
       end
Foo

julia> Foo.Module
Foo.Module

julia> @which Module
Core
```

What's so special about `Type`? I can just go ahead and call my type `LLVMType`, but using the module namespace for this seemed like a nicer approach.

### PR DESCRIPTION
**Pull request by maleadt**
This allows defining a type named 'Type' in packages.

### CODE REVIEW

**Comment by maleadt** [comment]
AppVeyor failure seems unrelated, restarted:

```
  Got an exception of type LoadError outside of a @test
  LoadError: Cannot clone ColorTypes from https://github.com/JuliaGraphics/ColorTypes.jl.git. Failed to receive response: The server returned an invalid or unrecognized response
  
   in prefetch at .\pkg\cache.jl:56
   in resolve at .\pkg\entry.jl:512
   in resolve at .\pkg\entry.jl:476
   in edit at .\pkg\entry.jl:30
   in #2 at .\task.jl:363
   in print_matrix at .\show.jl:1462
   in rehash! at .\dict.jl:585
   in rehash! at .\dict.jl:597
   in with at .\libgit2\types.jl:640
   in vect at .\abstractarray.jl:21
  while loading C:\projects\julia\test\pkg.jl, in expression starting on line 62
```

**Review by vtjnash — APPROVED** [review]

File: test/parse.jl [hunk 1]
```diff
@@ -811,3 +811,9 @@ end
 ```.head == :if
 
 end
+
+# issue 18756
+module Mod18756
+type Type
+end
+end
```
Comment (tkelman) [inline test/parse.jl]: actually test something about the result more than just not erroring?

Comment (maleadt) [inline test/parse.jl]: Sure; added a test.

**Comment by JeffBezanson** [comment]
:+1:

**Pull request merged**